plugins/nf-wave/src/main/io/seqera/wave/plugin/WaveClient.groovy

```diff
@@ -441,6 +544,81 @@ class WaveClient {
+    // https://github.com/spack/spack/blob/develop/share/spack/templates/container/Dockerfile
+    // LICENSE APACHE 2.0
+    protected String spackRecipeToDockerFile(String recipe) {
+
+        def noChecksumString = config.spackOpts().noChecksum ? '-n ' : ''
+        def result = """\
+# Builder image
+FROM ${config.spackOpts().builderImage} as builder
+
+RUN mkdir -p /opt/spack-env \\
+&&  sed -e 's;compilers:;compilers::;' \\
+         -e 's;^ *flags: *{};    flags:\\n      cflags: ${config.spackOpts().cFlags}\\n      cxxflags: ${config.spackOpts().cxxFlags}\\n      fflags: ${config.spackOpts().fFlags};' \\
+         /root/.spack/linux/compilers.yaml > /opt/spack-env/compilers.yaml \\
+&&  echo -e "\\
+spack: \\n\\
+  include: [/opt/spack-env/compilers.yaml] \\n\\
+  concretizer: \\n\\
+    unify: true \\n\\
+  specs: [${recipe}] \\n\\
+  packages: \\n\\
+    all: \\n\\
+      target: [${config.spackOpts().target}] \\n\\
+  config: \\n\\
+    install_tree: /opt/software \\n\\
+  view: /opt/view \\n\\
+" > /opt/spack-env/spack.yaml
+
+# Install packages, clean afterwards
+RUN cd /opt/spack-env && spack env activate . && spack install --fail-fast ${noChecksumString}&& spack gc -y
+
+# Strip binaries
+RUN find -L /opt/._view/* -type f -exec readlink -f '{}' \\; | \\
+    xargs file -i | \\
+    grep 'charset=binary' | \\
+    grep 'x-executable\\|x-archive\\|x-sharedlib' | \\
+    awk -F: '{print \$1}' | xargs strip -s
+
+RUN cd /opt/spack-env && \\
+    spack env activate --sh -d . >> /etc/profile.d/z10_spack_environment.sh && \\
+    original_view=\$( cd /opt ; ls -1d ._view/* ) && \\
+    sed -i "s;/view/;/\$original_view/;" /etc/profile.d/z10_spack_environment.sh && \\
+    echo "# Needed for Perl applications" >>/etc/profile.d/z10_spack_environment.sh && \\
+    echo "export PERL5LIB=\$(eval ls -d /opt/._view/*/lib/5.*):\$PERL5LIB" >>/etc/profile.d/z10_spack_environment.sh && \\
+    rm -rf /opt/view
+
+# Runner image
+FROM ${config.spackOpts().runnerImage}
+
+COPY --from=builder /opt/spack-env /opt/spack-env
+COPY --from=builder /opt/software /opt/software
+COPY --from=builder /opt/._view /opt/._view
+COPY --from=builder /etc/profile.d/z10_spack_environment.sh /etc/profile.d/z10_spack_environment.sh
+
+# Near OS-agnostic package addition
+RUN ( apt update -y && apt install -y procps libgomp1 ${config.spackOpts().osPackages} && rm -rf /var/lib/apt/lists/* ) || \\
+    ( yum install -y procps libgomp ${config.spackOpts().osPackages} && yum clean all && rm -rf /var/cache/yum ) || \\
+    ( zypper ref && zypper install -y procps libgomp1 ${config.spackOpts().osPackages} && zypper clean -a ) || \\
+    ( apk update && apk add --no-cache procps libgomp bash ${config.spackOpts().osPackages} && rm -rf /var/cache/apk )
+""" //.stripIndent()
+
+        result = addCommands(result)
+
+        result += """\
+
+ENTRYPOINT ["/bin/bash", "--rcfile", "/etc/profile", "-l", "-c", "\$*", "--" ]
+CMD [ "/bin/bash" ]
+"""//.stripIndent()
```

Comment:
also this it's better to keep into a template

Reply:
agree, it's in my TODO

Reply:
done